spec/vehicle_factory_spec.rb

```diff
@@ -0,0 +1,53 @@
+require 'spec_helper'
+
+
+RSpec.describe VehicleFactory do
+  it "exists" do
+    factory_1 = VehicleFactory.new
+    expect(factory_1).to be_a VehicleFactory
+  end
+
+  it "has a create_vehicles method" do
+    factory_1 = VehicleFactory.new
+    wa_ev_registrations = DmvDataService.new.wa_ev_registrations
+    expect(factory_1.create_vehicles(wa_ev_registrations)).to be_a Array
+  end
+
+  it "has a most popular make/model attribute" do
+    factory_1 = VehicleFactory.new
+    wa_ev_registrations = DmvDataService.new.wa_ev_registrations
+  
+    expect(factory_1.most_popular_make_model).to eq "" 
+    factory_1.create_vehicles(wa_ev_registrations)
+    expect(factory_1.most_popular_make_model).to eq("NISSAN, Leaf") 
+  end
+
+  it "has most popular year attribute" do
+    factory_1 = VehicleFactory.new
+    wa_ev_registrations = DmvDataService.new.wa_ev_registrations
+    factory_1.create_vehicles(wa_ev_registrations)
+  
+    expect(factory_1.most_common_year).to eq(2021)
+    expect(factory_1.most_common_year).not_to eq(2020)
+    expect(factory_1.year_counter(2015)).to eq(81)
+  end
+
+  it "has a method for most common county" do
+    factory_1 = VehicleFactory.new
+    wa_ev_registrations = DmvDataService.new.wa_ev_registrations
+    factory_1.create_vehicles(wa_ev_registrations)
+    
+    expect(factory_1.most_common_county).to eq("King")
+  end
+
+  it "has capability to use NY vehicle database" do
+    factory_2 = VehicleFactory.new
+    new_york_vehicle_registrations = DmvDataService.new.ny_state_registrations
+    factory_2.create_vehicles(new_york_vehicle_registrations)
+    
+    expect(factory_2.most_common_county).to eq("ERIE")
+    expect(factory_2.most_common_year).to eq(1949)
+    expect(factory_2.year_counter(1936)).to eq(9)
+    expect(factory_2.year_counter(1999)).to eq(4)
+  end
```

Comment:
These tests would be a great time to use fixture data. We haven't talked about this yet, but will briefly touch on it in the mocks/stubs class next week. Since the data that is returned from the API is going to change eventually, it's very possible that these tests will no longer pass. After about a year, it's likely that the most_common_year  or year_counter will change. So, for that reason, we could create our own subset of data that won't change, just to be sure that our methods work. What I'd do for that is create a `fixtures` folder in your spec folder, and in there, house a few files of data. Then, instead of using the `DmvDataService.new.wa_ev_registrations` data, you could use the data from the fixture file.